ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/JsonRpcHttpService.java

```diff
@@ -161,6 +169,29 @@ private void validateConfig(final JsonRpcConfiguration config) {
     return resultFuture;
   }
 
+  private Handler<RoutingContext> checkWhitelistHostHeader() {
+    return event -> {
+      Optional<String> hostHeader =
+          Optional.ofNullable(Iterables.get(Splitter.on(':').split(event.request().host()), 0));
+      if (config.hostsWhitelist().contains("*")
+          || (hostHeader.isPresent()
+              && config
+                  .hostsWhitelist()
+                  .stream()
+                  .anyMatch(
+                      whitelistEntry ->
+                          whitelistEntry.toLowerCase().equals(hostHeader.get().toLowerCase())))) {
```

Comment:
It would probably be helpful to extract the second part of this or condition to a `hostIsInWhitelist` method.